src/bootstrap/builder.rs

```diff
@@ -898,6 +902,11 @@ impl<'a> Builder<'a> {
             extra_args.push_str(&s);
         }
 
+        if cmd == "clippy" {
+            extra_args.push_str("-Zforce-unstable-if-unmarked -Zunstable-options \
```

Comment:
have you tested that this prevents the rebuilds? I would have assumed this to still cause build thrashing. If this works, then I'm happy with it, if not, you should be able to add them in the `args` function you created below by putting them after a `--` argument

Reply:
I just checked it - there are no rebuilds. The only thing is that after throwing its warnings (a lot of stuff found in `libcore`) `./x.py clippy` is complaining that it `can't find crate for std` and I no longer remember if this was an acceptable step forward or not ^^. I remember having trouble making it respect the build order.

Reply:
While not great, we can solve this in follow-ups. I think this may happen whenever one tries to run clippy on a target without libstd, so it's likely that it's a clippy issue.